File: src/main/java/pt/joaomneto/titancompanion/adventure/impl/TROKAdventure.java
package pt.joaomneto.titancompanion.adventure.impl;

import android.os.Bundle;
import android.view.Menu;

import java.io.BufferedWriter;
import java.io.IOException;

import pt.joaomneto.titancompanion.R;
import pt.joaomneto.titancompanion.adventure.Adventure;

public class TROKAdventure extends Adventure {

	protected static final int FRAGMENT_VEHICLE_COMBAT = 2;
	protected static final int FRAGMENT_EQUIPMENT = 3;
	protected static final int FRAGMENT_NOTES = 4;
	private int currentWeapons = -1;
	private int currentShields = -1;
	private int initialWeapons = -1;
	private int initialShields = -1;
	private int missiles = -1;

	public TROKAdventure() {
		super();
		fragmentConfiguration.clear();
		fragmentConfiguration.put(FRAGMENT_VITAL_STATS, new AdventureFragmentRunner(R.string.vitalStats, "pt.joaomneto.titancompanion.adventure.impl.fragments.AdventureVitalStatsFragment"));
		fragmentConfiguration.put(FRAGMENT_COMBAT, new AdventureFragmentRunner(R.string.fights, "pt.joaomneto.titancompanion.adventure.impl.fragments.trok.TROKAdventureCombatFragment"));
		fragmentConfiguration.put(FRAGMENT_VEHICLE_COMBAT, new AdventureFragmentRunner(R.string.starshipCombat, "pt.joaomneto.titancompanion.adventure.impl.fragments.trok.TROKStarShipCombatFragment"));
		fragmentConfiguration.put(FRAGMENT_EQUIPMENT, new AdventureFragmentRunner(R.string.moneyEquipment, "pt.joaomneto.titancompanion.adventure.impl.fragments.AdventureEquipmentFragment"));
		fragmentConfiguration.put(FRAGMENT_NOTES, new AdventureFragmentRunner(R.string.notes, "pt.joaomneto.titancompanion.adventure.impl.fragments.AdventureNotesFragment"));
	}

	@Override
	protected void onCreate(Bundle savedInstanceState) {
		try {
			super.onCreate(savedInstanceState);

		} catch (Exception e) {
			e.printStackTrace();
		}

	}

	@Override
	public boolean onCreateOptionsMenu(Menu menu) {
		getMenuInflater().inflate(R.menu.adventure, menu);
		return true;
	}

	@Override
	public void storeAdventureSpecificValuesInFile(BufferedWriter bw) throws IOException {

		bw.write("currentWeapons=" + currentWeapons + "\n");
		bw.write("currentShields=" + currentShields + "\n");
		bw.write("initialWeapons=" + initialWeapons + "\n");
		bw.write("initialShields=" + initialShields + "\n");
		bw.write("missiles=" + missiles + "\n");
		bw.write("gold=" + getGold() + "\n");
		bw.write("provisions=" + getProvisions() + "\n");
		bw.write("provisionsValue=" + getProvisionsValue() + "\n");
	}

	public int getCurrentWeapons() {
		return currentWeapons;
	}

	public void setCurrentWeapons(int currentWeapons) {
		this.currentWeapons = currentWeapons;
	}

	public int getCurrentShields() {
		return currentShields;
	}

	public void setCurrentShields(int currentShields) {
		this.currentShields = currentShields;
	}

	public int getInitialWeapons() {
		return initialWeapons;
	}

	public void setInitialWeapons(int initialWeapons) {
		this.initialWeapons = initialWeapons;
	}

	public int getInitialShields() {
		return initialShields;
	}

	public void setInitialShields(int initialShields) {
		this.initialShields = initialShields;
	}

	@Override
	protected void loadAdventureSpecificValuesFromFile() {

		setCurrentWeapons(Integer.valueOf(getSavedGame().getProperty("currentWeapons")));
		setCurrentShields(Integer.valueOf(getSavedGame().getProperty("currentShields")));
		setInitialWeapons(Integer.valueOf(getSavedGame().getProperty("initialWeapons")));
		setInitialShields(Integer.valueOf(getSavedGame().getProperty("initialShields")));
		setMissiles(Integer.valueOf(getSavedGame().getProperty("missiles")));
		setProvisions(Integer.valueOf(getSavedGame().getProperty("provisions")));
		setProvisionsValue(Integer.valueOf(getSavedGame().getProperty("provisionsValue")));
		setGold(Integer.valueOf(getSavedGame().getProperty("gold")));

	}

	public int getMissiles() {
		return missiles;
	}

	public void setMissiles(int missiles) {
		this.missiles = missiles;
	}

	public String getConsumeProvisionText() {
		return getResources().getString(R.string.usePepPill);
	}
	

	public String getProvisionsText() {
		return getResources().getString(R.string.pepPills);
	}
	
	public String getCurrencyName() {
		return getResources().getString(R.string.kopecks);
	}

}

File: src/main/java/pt/joaomneto/titancompanion/adventure/impl/TROKAdventure.kt
package pt.joaomneto.titancompanion.adventure.impl

import android.os.Bundle
import android.view.Menu
import pt.joaomneto.titancompanion.R
import pt.joaomneto.titancompanion.adventure.Adventure
import pt.joaomneto.titancompanion.adventure.impl.fragments.AdventureEquipmentFragment
import pt.joaomneto.titancompanion.adventure.impl.fragments.AdventureNotesFragment
import pt.joaomneto.titancompanion.adventure.impl.fragments.AdventureVitalStatsFragment
import pt.joaomneto.titancompanion.adventure.impl.fragments.trok.TROKAdventureCombatFragment
import pt.joaomneto.titancompanion.adventure.impl.fragments.trok.TROKStarShipCombatFragment
import pt.joaomneto.titancompanion.util.AdventureFragmentRunner
import java.io.BufferedWriter
import java.io.IOException

class TROKAdventure : Adventure(
    arrayOf(
        AdventureFragmentRunner(R.string.vitalStats, AdventureVitalStatsFragment::class),
        AdventureFragmentRunner(R.string.fights, TROKAdventureCombatFragment::class),
        AdventureFragmentRunner(R.string.starshipCombat, TROKStarShipCombatFragment::class),
        AdventureFragmentRunner(R.string.moneyEquipment, AdventureEquipmentFragment::class),
        AdventureFragmentRunner(R.string.notes, AdventureNotesFragment::class)
    )
) {
    var currentWeapons = -1
    var currentShields = -1
    var initialWeapons = -1
    var initialShields = -1
    var missiles = -1

    override val consumeProvisionText = R.string.usePepPill

    override val provisionsText = R.string.pepPills

    override val currencyName = R.string.kopecks

    override fun onCreate(savedInstanceState: Bundle?) {
        try {
            super.onCreate(savedInstanceState)
        } catch (e: Exception) {
            e.printStackTrace()
        }
    }

    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        menuInflater.inflate(R.menu.adventure, menu)
        return true
    }

    @Throws(IOException::class)
    override fun storeAdventureSpecificValuesInFile(bw: BufferedWriter) {

        bw.write("currentWeapons=" + currentWeapons + "\n")
        bw.write("currentShields=" + currentShields + "\n")
        bw.write("initialWeapons=" + initialWeapons + "\n")
        bw.write("initialShields=" + initialShields + "\n")
        bw.write("missiles=" + missiles + "\n")
        bw.write("gold=" + gold + "\n")
        bw.write("provisions=" + provisions + "\n")
        bw.write("provisionsValue=" + provisionsValue + "\n")
    }

    override fun loadAdventureSpecificValuesFromFile() {

        currentWeapons = Integer.valueOf(savedGame.getProperty("currentWeapons"))
        currentShields = Integer.valueOf(savedGame.getProperty("currentShields"))
        initialWeapons = Integer.valueOf(savedGame.getProperty("initialWeapons"))
        initialShields = Integer.valueOf(savedGame.getProperty("initialShields"))
        missiles = Integer.valueOf(savedGame.getProperty("missiles"))
        provisions = Integer.valueOf(savedGame.getProperty("provisions"))
        provisionsValue = Integer.valueOf(savedGame.getProperty("provisionsValue"))
        gold = Integer.valueOf(savedGame.getProperty("gold"))
    }

    companion object {

        protected val FRAGMENT_VEHICLE_COMBAT = 2
        protected val FRAGMENT_EQUIPMENT = 3
        protected val FRAGMENT_NOTES = 4
    }
}
